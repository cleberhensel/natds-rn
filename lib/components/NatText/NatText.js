import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from './NatText.styles';
import { ThemeNaturaLight } from '../../themes/ThemeNaturaLight';

export const NatText = ({
  value,
  type,
  color,
  margin,
}) => (
  <View margin={margin}>
    <Text type={type} color={color}>
      {value}
    </Text>
  </View>
);

NatText.propTypes = {
  color: PropTypes.string,
  margin: PropTypes.number,
  type: PropTypes.string,
  value: PropTypes.string,
};

NatText.defaultProps = {
  color: ThemeNaturaLight.palette.text.primary,
  margin: ThemeNaturaLight.spacing.spacingStandard,
  type: 'body1',
  value: '',
};
