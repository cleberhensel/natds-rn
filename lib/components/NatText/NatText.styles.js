import styled from 'styled-components/native';
import { fontSize } from '../../tokens';

const getStyles = (type, color) => Object.assign(fontSize[type], { color });

export const View = styled.View`
  padding-left: 0;
  padding-right: 0;
  padding-top: 0;
  padding-bottom: 0;
  margin-left:0;
  margin-right:0;
  margin-top:0;
  margin-bottom: ${({ margin }) => margin};
`;

export const Text = styled.Text`
  font-size: ${({ type, color }) => getStyles(type, color).fontSize};
  color: ${({ type, color }) => getStyles(type, color).color};
  line-height: ${({ type, color }) => getStyles(type, color).lineHeight};
  font-weight: ${({ type, color }) => getStyles(type, color).fontWeight};
`;
