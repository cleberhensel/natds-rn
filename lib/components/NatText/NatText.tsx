import React from "react";
import { View, Text } from "./NatText.styles";
import { ThemeNaturaLight } from "../../themes/ThemeNaturaLight";
type NatTextProps = {
  color?: string,
  margin?: number,
  type?: string,
  value?: string
};
export const NatText: React.SFC<NatTextProps> = ({
  value,
  type,
  color,
  margin
}) => (
  <View margin={margin}>
    <Text type={type} color={color}>
      {value}
    </Text>
  </View>
);
NatText.defaultProps = {
  color: ThemeNaturaLight.palette.text.primary,
  margin: ThemeNaturaLight.spacing.spacingStandard,
  type: "body1",
  value: ""
};
