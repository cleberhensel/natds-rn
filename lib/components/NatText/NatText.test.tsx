import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components/native";
import { NatText } from "./NatText";
describe("NatText component default", () => {
  it("Should render NatText component ", () => {
    const button = renderer.create(<NatText value="Teste" />).toJSON();
    expect(button).toMatchSnapshot("NatText snapshot");
  });
});
