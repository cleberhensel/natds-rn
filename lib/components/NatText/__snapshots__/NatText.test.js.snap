// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NatText component default Should render NatText component  1`] = `
<View
  margin={24}
  style={
    Array [
      Object {
        "marginBottom": 24,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      },
    ]
  }
>
  <Text
    color="#333333"
    style={
      Array [
        Object {
          "color": "#333333",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
      ]
    }
    type="body1"
  >
    Teste
  </Text>
</View>
`;
